modules/terrain/src/lib/parse-terrain.js

```diff
@@ -89,15 +109,49 @@ function getMartiniTileMesh(terrainImage, terrainOptions) {
       boundingBox: getMeshBoundingBox(attributes)
     },
     mode: 4, // TRIANGLES
-    indices: {value: triangles, size: 1},
+    indices: {value: Uint32Array.from(triangles), size: 1},
     attributes
   };
 }
 
+/**
+ * Get Martini generated vertices and triangles
+ *
+ * @param {number} meshMaxError threshold for simplifying mesh
+ * @param {number} width width of the input data
+ * @param {*} terrain elevation data
+ * @returns generated vertices and triangles
+ */
+function getMartiniTileMesh(meshMaxError, width, terrain) {
+  const gridSize = width + 1;
+  const martini = new Martini(gridSize);
+  const tile = martini.createTile(terrain);
+  const {vertices, triangles} = tile.getMesh(meshMaxError);
+
+  return {vertices, triangles};
+}
+
+/**
+ * Get Delatin generated vertices and triangles
+ *
+ * @param {number} meshMaxError threshold for simplifying mesh
+ * @param {number} width width of the input data array
+ * @param {number} height height of the input data array
+ * @param {*} terrain elevation data
+ * @returns generated vertices and triangles
```

Comment:
Ideally we'd have a typed return object, where we declare the interface. In this case

```
vertices: number[]
triangles: number[]
```

but in the martini case
```
vertices: Uint16Array
triangles: Uint32Array
```

Not sure the best way to define these interfaces using jsdoc typing.

Reply:
I changed it to returning an object containing the vertices and triangles data